integral_view/views/disk_management.py

```diff
@@ -50,69 +50,72 @@ def view_disks(request):
-                initial['path'] = request.GET['path']
-                return_dict['path'] = request.GET['path']
-                return_dict['disk_id'] = request.GET['disk_id']
-                target = None
-                form = zfs_forms.FormatDiskForm(initial = initial,  datasets = datasets)
-            
-                return_dict['form'] = form
-                return django.shortcuts.render_to_response('format_disks.html', return_dict, context_instance = django.template.context.RequestContext(request))
+        if "path" in request.GET:
+            path = request.REQUEST['path']
+            total_partitions = request.REQUEST["total_partitions"]
+            if request.GET["total_partitions"] == '0':
+                output, err = disks.unmount_disk_by_name(path)
+                if err:
+                    raise Exception(err)
+
+                return django.http.HttpResponseRedirect('/view_disks?ack=unmount_disk')
     except Exception, e:
         return_dict['base_template'] = "storage_base.html"
         return_dict["page_title"] = 'Disks'
         return_dict['tab'] = 'view_disks_tab'
-        return_dict["error"] = 'Error getting partition type for formatting disk'
+        return_dict["error"] = 'Error mounting/unmounting disk'
         return_dict["error_details"] = str(e)
         return django.shortcuts.render_to_response("logged_in_error.html", return_dict, context_instance=django.template.context.RequestContext(request))
 
 
-
-def format_complete(request):
+def format_disk(request):
     return_dict = {}
     try:
-        if "f_type" in request.GET:
-            if 'path' in request.GET:
-                output, err = disks.format_usb_disk_by_name(request.GET['path'],request.GET['f_type'])
+        if request.method == 'GET':
+            if 'disk_id' in request.GET:
+                fs_types, err = disks.get_supported_mkfs_file_systems()
+                if err:
+                    raise Exception(err)
+                initial = {}
+                initial['path'] = request.GET['path']
+                return_dict['path'] = request.GET['path']
+                return_dict['disk_id'] = request.GET['disk_id']
+                target = None
+                form = disk_forms.FormatDiskForm(initial = initial,  fs_types = fs_types)
+                return_dict['form'] = form
+                return django.shortcuts.render_to_response('format_disks.html', return_dict, context_instance = django.template.context.RequestContext(request))
+        if request.method == 'POST':
+            if 'disk_id' in request.POST:
+                return_dict['fs_type'] = request.POST['fs_type']
+                return_dict['disk_id'] = request.POST['disk_id']
+                return_dict['path'] = request.POST['path']
+                return django.shortcuts.render_to_response('format_disk_conf.html', return_dict, context_instance = django.template.context.RequestContext(request))
+            if 'fs_type' in request.POST:
```

Comment:
I am a bit confused about this.. How do you know if the request has come from a confirmation page or not?